protocol-designer/src/components/StepEditForm/fields/WellOrderField/WellOrderModal.tsx

```diff
@@ -119,136 +113,134 @@ export class WellOrderModal extends React.Component<
     }
   }
 
-  applyChanges: () => void = () => {
-    this.props.updateValues(this.state.firstValue, this.state.secondValue)
+  const [state, setState] = React.useState<State>({
+    firstValue: DEFAULT_FIRST,
+    secondValue: DEFAULT_SECOND,
+  })
+
+  React.useEffect(() => {
+    const { firstValue, secondValue } = props
+    if (firstValue != null && secondValue != null) {
+      setState({
+        firstValue: firstValue,
+        secondValue: secondValue,
+      })
+    }
+  }, [props])
```

Comment:
I think specifying dependencies would be safe since we can avoid unexpected update.

```suggestion
  }, [firstValue, secondValue])
```

Reply:
oops yes good catch thank you!